core/controllers/editor.py

```diff
@@ -960,30 +962,49 @@ def post(self, exploration_id):
                 'Invalid filename extension: it should have '
                 'one of the following extensions: %s' % allowed_formats)
 
-        temp_suffix = '.' + extension
-        with tempfile.NamedTemporaryFile(suffix=temp_suffix) as f:
-            f.write(raw)
-            try:
-                audio = mutagen.File(f.name)
-            except mutagen.MutagenError:
-                raise self.InvalidInputException('Audio not recognized '
-                                                 'as a %s file' % extension)
-            f.close()
-
+        tempbuffer = StringIO.StringIO()
+        tempbuffer.write(raw)
+        tempbuffer.seek(0)
+        try:
+            audio = mutagen.File(tempbuffer)
+        except mutagen.MutagenError:
+            # Mutagen occasionally has problems with certain files
```

Comment:
For future reference, could you add a sample Apache-shareable audio file (that does trigger this error) in core/tests/data, and include a reference to that file in this comment? I can see this being useful in the future.